src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/AuthorizationStateRetrieverService.kt

```diff
@@ -45,4 +55,68 @@ class AuthorizationStateRetrieverService(
         }
       })
   }
+
+  fun getOrder(
+    trx: BaseTransactionWithRequestedAuthorization,
+  ): Mono<OrderResponseDto> {
+    return Mono.just(trx)
+      .filter { transaction ->
+        transaction.transactionAuthorizationRequestData.paymentGateway ==
+          TransactionAuthorizationRequestData.PaymentGateway.NPG
+      }
+      .switchIfEmpty {
+        logger.info(
+          "Trying get order for invalid payment gateway: [{}]",
+          trx.transactionAuthorizationRequestData.paymentGateway.name)
+        Mono.error(InvalidNPGPaymentGatewayException(trx.transactionId))
+      }
+      .flatMap { transaction ->
+        val activationData =
+          (transaction.transactionActivatedData.transactionGatewayActivationData
+            as NpgTransactionGatewayActivationData)
+        getOrder(
+          transactionId = trx.transactionId,
+          pspId = transaction.transactionAuthorizationRequestData.pspId,
+          paymentMethod =
+            NpgClient.PaymentMethod.valueOf(
+              transaction.transactionAuthorizationRequestData.paymentMethodName),
+          orderId = activationData.orderId,
+          correlationId = activationData.correlationId)
+      }
+  }
+
+  private fun getOrder(
```

Comment:
I don't like having two `getOrder` functions that do different things, wdyt?

```suggestion
  private fun performGetOrder(
```

Reply:
Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/269/commits/5e82102aa8a210c05b4ded48ac71a3a5c518a23b